os/fs/smartfs/smartfs_utils.c

```diff
@@ -1321,7 +1291,8 @@ int smartfs_writeentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s new_
 		SET_ENTRY_FLAGS(new_entry.flags, mode, type);
 	}
 
-	if (newchain) {
+	/* If passed parameter parentdirsector != new_entry.dsector, it means it is a new chain sector that we will write to */
+	if (parentdirsector != psector) {
```

Comment:
line 1309, below is correct, am I right?
chainheader->type = type;

Reply:
Type parameter passed is for the type of the new entry that is to be written.

Chainheader is for writing the chainheader of the new sector that was allocated for chaining to the parent, so it will always have type = SMARTFS_DIRENT_TYPE_DIR

Reply:
yeah correct. =)